packages/@aws-cdk/core/lib/asset-staging.ts

```diff
@@ -144,6 +144,9 @@ export class AssetStaging extends Construct {
 
     // Create temp directory for bundling inside the temp staging directory
     const bundleDir = path.resolve(fs.mkdtempSync(path.join(stagingTmp, 'asset-bundle-')));
+    // Chmod the bundleDir to full access after applying the process the umask.
+    // tslint:disable-next-line:no-bitwise
+    fs.chmodSync(bundleDir, 0o777 & (~process.umask()));
```

Comment:
Looks like calling `process.umask()` without no argument is now deprecated in Node.js v14: https://nodejs.org/docs/latest-v14.x/api/process.html#process_process_umask

What would be the alternative? What about using only `Oo777`, would it be OK?

Reply:
Another alternative is to set a new umask (to get the return value), then restore the original umask. I've seen a [getumask manpage](https://man7.org/linux/man-pages/man3/getumask.3.html) that describes this approach. But, `0o777` would certainly satisfy my use case. I can't think of any major downsides. What's your preference?

Reply:
> But, mode `0o777` alone would certainly satisfy my use case. I can't think of any major downsides.

OK, let's go for this.